chromadb/segment/impl/metadata/grpc_segment.py

```diff
@@ -96,16 +98,24 @@ def get_metadata(
             limit=limit,
             offset=offset,
             include_metadata=include_metadata,
+            request_metadata=request_metadata,
         )
         limit = limit or 2**63 - 1
         offset = offset or 0
 
         if limit and limit < 0:
             raise ValueError("Limit cannot be negative")
+        try:
+            response: pb.QueryMetadataResponse = (
+                self._metadata_reader_stub.QueryMetadata(
+                    request, timeout=self._request_timeout_seconds
+                )
+            )
+        except grpc.RpcError as e:
+            if e.details() == "Collection version mismatch":
+                raise ValueError("Collection version mismatch")
```

Comment:
i think i'd prefer a new error type so we don't string match

Reply:
Sounds good. Found errors.py.